db/upgrade.php

```diff
@@ -987,5 +987,65 @@ function xmldb_local_recompletion_upgrade($oldversion) {
         upgrade_plugin_savepoint(true, 2023112600, 'local', 'recompletion');
     }
 
+    if ($oldversion < 2024071100) {
+        // We renamed some site admin settings.
```

Comment:
can we create an array with the list of oldname -> newname and then pass it through a foreach loop that calls the work required for this?

Also - we need to use set_config() - not a sql call to update config_plugins entries because config values are  cached. (even if we could use sql, you're missing the "where pluginname = 'recompletion" in the sql :-)

eg 
```
$update = ['choice' => 'choiceattempts', 'customcert' => 'customcertificates''];
foreach ($update as $new -> $old {
    set_config()
}